File: question/leetcode/leetcode3576/leetcode3576.py
class Solution:
    def canMakeEqual(self, nums: List[int], k: int) -> bool:
        def f(target):
            arr = nums[:]  # 不修改原数组
            n = len(arr)
            count = 0
            for i in range(n - 1):
                if arr[i] != target:
                    # 操作：翻转当前和下一个
                    arr[i] *= -1
                    arr[i + 1] *= -1
                    count += 1
                    if count > k:
                        return False
            return all(x == target for x in arr)
        
        return f(1) or f(-1)

File: question/leetcode/leetcode3576/leetcode3576.go
package main

func canMakeEqual(nums []int, k int) bool {
	return tryTarget(nums, 1, k) || tryTarget(nums, -1, k)
}

func tryTarget(nums []int, target int, k int) bool {
	arr := make([]int, len(nums))
	copy(arr, nums)
	count := 0
	for i := 0; i < len(arr)-1; i++ {
		if arr[i] != target {
			arr[i] *= -1
			arr[i+1] *= -1
			count++
			if count > k {
				return false
			}
		}
	}
	// 最后一位不能翻转
	return arr[len(arr)-1] == target
}
